### PR TITLE
Make template required in krane-run

### PR DESCRIPTION
**Pull request by jessie-JNing**
**What are you trying to accomplish with this PR?**
close: https://github.com/Shopify/kubernetes-deploy/issues/568

Make `--template` to be a required argument in krane-run.

**How is this accomplished?**
- change `template` to be required in run cli
-  add `with_processed_task_template` check in exe command, raise `OptionsError` if missing.
- fix tests
- update readme

### CODE REVIEW

**Comment by jessie-JNing** [comment]
> Isn't the validation in `RunnerTaskConfigValidator` already covering this case? If that's true, we shouldn't need the change in `OptionsHelper`. Example:

I consider about these two validations: 
`RunnerTaskConfigValidator` validate the template given, if it's an empty string to invalid path. 
The `OptionsHelper` should check if `--template` arg is given or not, so update the code.  
@dirceu could you review  👀it?

File: lib/kubernetes-deploy/options_helper.rb [hunk 1]
```diff
@@ -29,6 +29,14 @@ def with_processed_template_paths(template_paths, require_explicit_path: false)
         end
       end
 
+      def with_processed_task_template(template_path)
```
Comment (dirceu) [inline lib/kubernetes-deploy/options_helper.rb]: `OptionsHelper.with_processed_template_paths` both manipulates **and** validates the value(s) we pass to it, so its block interface makes sense.

In this particular case we're just validating the value so maybe we could change the name to reflect that. What do you think?

Also, we maybe don't need the block interface, but I have no strong feelings about that.

File: exe/kubernetes-run [hunk 2]
```diff
@@ -5,7 +5,7 @@ require 'kubernetes-deploy/runner_task'
 require 'kubernetes-deploy/options_helper'
 require 'optparse'
 
-template = "task-runner-template"
+template = nil
```
Comment (dturn) [inline exe/kubernetes-run]: The `exe/kubernetes-*` tasks need to remain backwards compatible.

File: lib/kubernetes-deploy/options_helper.rb (hunk 1 above)
Comment (jessie-JNing) [reply lib/kubernetes-deploy/options_helper.rb]: Thanks for the comment @dirceu , I remove the check `OptionsHelper.with_processed_template_paths` as it should be backwards compatible: https://github.com/Shopify/kubernetes-deploy/pull/579#discussion_r332127566

File: exe/kubernetes-run (hunk 2 above)
Comment (jessie-JNing) [reply exe/kubernetes-run]: Updated the PR.  `exe/kubernetes-run` remains as it is. The `--template` is changed to be required in CLI.

File: test/exe/run_test.rb [hunk 3]
```diff
@@ -73,9 +82,10 @@ def set_krane_run_expectations(new_args: {}, run_args: {})
   end
 
   def krane_run!(flags: '')
+    cmd_flags = "#{flags} --template #{TASK_TEMPLATE}".split
```
Comment (dturn) [inline test/exe/run_test.rb]: Unconditionally adding `--template #{TASK_TEMPLATE}` seems wrong. I liked the pattern from the deploy test (` flags += ' -f /tmp' unless flags.include?('-f')`)

File: test/exe/run_test.rb [hunk 4]
```diff
@@ -42,22 +44,29 @@ def test_run_parses_env_vars
     krane_run!(flags: '--env-vars SOMETHING=8000,FOO=bar')
   end
 
+  def test_run_failure_with_no_template_given
```
Comment (dturn) [inline test/exe/run_test.rb]: I don't think we need to test that thor 'require' works as expected.

File: test/exe/run_test.rb (hunk 3 above)
Comment (jessie-JNing) [reply test/exe/run_test.rb]: Thanks Danny, that pattern is really a more correct way. Will use it

**Pull request merged**